src/App.tsx

```diff
@@ -1,31 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
+export const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
+const total = 42;
+
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const startPoint = currentPage * perPage - perPage;
+  const endPoint = currentPage * perPage;
+  const options = [3, 5, 10, 20];
+
+  function getEnd(value: number) {
+    let end = value;
+
+    if (value > total) {
+      end = total;
+    }
+
+    return end;
+  }
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${startPoint + 1} - ${getEnd(endPoint)} of ${total})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
-            id="perPageSelector"
             className="form-control"
+            value={perPage}
+            onChange={(event) => {
+              setPerPage(Number(event.target.value));
+              setCurrentPage(1);
+            }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here

**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**